src/main/java/Duke.java

```diff
@@ -1,10 +1,362 @@
+                    addToList(t, Tasks, Tasks.size());
+                    printTask(t, Tasks.size());
+                } else if (str.contains("event")) {
+                    checkEventString(str);
+                    checkFormat(str);
+                    String time = str.substring(str.indexOf("/") + 3, str.length());
+                    if (str.length() - 6 + 11 > maxlength) maxlength = str.length() - 6 + 11;
+                    Event t = new Event(str.substring(6, str.indexOf("/") - 1), time);
+                    addToList(t, Tasks, Tasks.size());
+                    printTask(t, Tasks.size());
+                } else if (str.contains("delete")) {
+                    String numberOnly = str.replaceAll("[^0-9]", "");
+                    if(numberOnly.length() == 0) numberOnly = "0"; // to avoid empty string error
+                    int num = Integer.parseInt(numberOnly);
+                    checkNum(num, Tasks.size());
+                    Task t = Tasks.get(num - 1);
+                    Tasks.remove(num - 1);
+                    printDelete(t, Tasks.size());
+                } else if (str.contains("done")) {
+                    String numberOnly = str.replaceAll("[^0-9]", "");
+                    if(numberOnly.length() == 0) numberOnly = "0"; // to avoid empty string error
+                    int num = Integer.parseInt(numberOnly);
+                    checkNum(num, Tasks.size());
+                    Tasks.get(num - 1).setDone(true);
+                    printDone(Tasks.get(num - 1));
+                } else {
+                    FormatError();
+                }
+            } catch (AssignException e){
+                  printString("OOPS! not assigned");
+            } catch (DeadlineStringException e){
+                  printString("OOPS! the deadline description cannot be empty");
+            } catch (EventStringException e) {
+                  printString("OOPS! the event description cannot be empty");
+            } catch (FormatException e) {
+                  printString("OOPS! what's the time?");
+            } catch (TodoStringException e) {
+                  printString("OOPS! the todo description cannot be empty");
+            } catch (TypingException e) {
+                  printString("OOPS! what do u mean?");
+            } catch (EmptyListException e) {
+                  printString("OOPS! empty list");
+            }
+        }
+
+        try {
+            FileWriter fileWriter =new FileWriter(file);
+            fileWriter.write("");
+            fileWriter.flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        String textToAdd = "";
+        for(int i = 0; i < Tasks.size(); i++){
+            switch (Tasks.get(i).getType()){
+            case TODO:
+                textToAdd += "T | " + (Tasks.get(i).isDone() ? "1" : "0") + " | " + Tasks.get(i).getDescription() + System.lineSeparator();
```

Comment:
This line exceeds the conventional length of [<120 characters per line](https://se-education.org/guides/conventions/java/basic.html#layout). There are a few lines that are also doing the same thing. Perhaps reformatting the string with [Java String format() method](https://www.javatpoint.com/java-string-format) would help.